src/cmd-cloud-prune

```diff
@@ -0,0 +1,291 @@
+        bucket, prefix = builds_source_data_url.split('//', 1)[1].split('/', 1)
+        prefix = prefix[:-len("/builds")]
+        return bucket, prefix
+    else:
+        raise Exception("Invalid scheme: only s3:// supported")
+
+
+def get_cloud_config(args):
+    return {
+        "gcp": {
+            "json-key": args.gcp_json_key,
+            "project": args.gcp_project
+        },
+        "aws": {
+            "credentials": args.aws_config_file
+        }
+    }
+
+
+def validate_policy(stream, policy):
+    # If the build key is set in the policy file, then the cloud-uploads key must
+    # also be present, and the duration of cloud-uploads must be equal or shorter
+    if "build" in policy[stream]:
+        actions = policy[stream]
+        if 'cloud-uploads' not in actions:
+            raise Exception("Pruning for cloud-uploads must be set before we prune the builds")
+        cloud_uploads_duration = get_period_in_months(actions["cloud-uploads"])
+        build_duration = get_period_in_months(actions["build"])
+        if cloud_uploads_duration > build_duration:
+            raise Exception("Duration of pruning cloud-uploads must be less than or equal to pruning a build")
+
+
+def get_supported_images(meta_json, unsupported, supported):
+    images = {}
+    for key in meta_json:
+        if key in unsupported:
+            raise Exception(f"The platform {key} is not supported")
+        if key in supported:
+            images[key] = meta_json[key]
+    return images
+
+
+def parse_fcos_version(version):
+    m = re.match(r'^([0-9]{2})\.([0-9]{8})\.([0-9]+)\.([0-9]+)$', version)
+    if m is None:
+        raise Exception(f"Incorrect versioning for FCOS build {version}")
+    try:
+        timestamp = datetime.datetime.strptime(m.group(2), '%Y%m%d')
+    except ValueError:
+        raise Exception(f"FCOS build {version} has incorrect date format. It should be in (%Y%m%d)")
+    return timestamp
+
+
+def get_json_from_s3(s3, bucket, key):
+    try:
+        response = s3.get_object(Bucket=bucket, Key=key)
+        content = response["Body"].read().decode("utf-8")
+        return json.loads(content)
+    except Exception as e:
+        raise Exception(f"Error fetching the JSON file from S3 {bucket}/{key}: {e}")
+
+
+def save_builds_json(builds_json_data):
+    with open("builds/builds.json", "w") as json_file:
+        json.dump(builds_json_data, json_file, indent=2)
+
+
+def handle_upload_builds_json(s3_client, bucket, prefix, dry_run, acl):
+    remote_builds_json_data = get_json_from_s3(s3_client, bucket, os.path.join(prefix, "builds/builds.json"))
+    with open(BUILDFILES['sourcedata'], "r") as f:
+        builds_json_source_data = json.load(f)
+    # Check if there are any changes that were made to remote(s3 version) builds.json
+    # while the pruning was in progress
+    if remote_builds_json_data != builds_json_source_data:
+        raise Exception("Detected remote modifications to builds.json")
```

Comment:
We can't error out here. We need to implement the merge logic.

Reply:
Yep. This was just temporary. Need further discussion on the merge logic.